app.py

```diff
@@ -24,25 +24,29 @@ def experience():
     '''
 
     if request.method == "GET":
-        return jsonify(data.get("experience", []))
+        return jsonify([edu.__dict__ for edu in data['experience']])
 
     if request.method == "POST":
-        post_data: dict[str, str] = request.get_json()
-        experiences = data.get("experience", [])
-        experiences.append(
-            Experience(
-                post_data["title"],
-                post_data["company"],
-                post_data["start_date"],
-                post_data["end_date"],
-                post_data["description"],
-                post_data["logo"],
-            )
-        )
-
-        return jsonify({"id": len(data.get("experience", [])) - 1})
-
-    return jsonify({})
+        required_fields = ['title', 'company', 'start_date', 'end_date', 'description', 'logo']
+        
+        if not request.json:
+            return jsonify({'error': 'No data provided'}), 400
+        
+        missing_fields = [field for field in required_fields if field not in request.json]
+        if missing_fields:
+            return jsonify({'error': f'Missing required fields: {", ".join(missing_fields)}'}), 400
```

Comment:
Generally, it is not a good idea to let a potential malicious user know what are the required fields and what are not. We could just say that some fields are missing, and not say what are those.
I don't think this necessary to do here, but just wanted to point out to ask what you think?